portality/scripts/anon_export.py

```diff
@@ -0,0 +1,129 @@
+import esprit
+from portality.core import app
+from portality.lib.anon import basic_hash, anon_name, anon_email
+
+
+def anonymise_email(record):
+    if 'email' in record:
+        record['email'] = anon_email(record['email'])
```

Comment:
I think overall this file is awkwardly structured and implemented, and comes with some maintenance overheads in the long run.  Because it uses the raw data rather than the model objects, it relies on an innate understanding of the underlying data model, which will break as soon as we change it.  Instead, each anonymising function should take a model object, and should perform the anonymisation by re-writing the properties via the object's API.  This will make this module easier to read and less likely to break or have bugs in it.

So, for example, the above should be something like

```
def anonymise_account_email(account):
    account.email = anon_email(account.email)
```

I realise this makes this function less re-usable for other objects that also have an `email` property, but given that all this does is check there is an email property and then call another function that anonymises the email address, that's not a loss.

In fact, I wouldn't even have this function at all, I'd just put the operational content into the `anonymise_account` function.

Reply:
Excellent point! I came at it from the perspective of what esprit gives me so didn't think of this. Esprit supplies the raw record to the transform function (which is what all these are). At the moment, it doesn't pass on the type of the record it's dumping to the transform f(), so the tranform wouldn't know which model class to initialise with the incoming esprit record data. I could modify esprit so it passes the type on to the transform f(). Then that will be able to init a model and do what you suggest.

Reply:
I'm not sure why it's hard to get models, we know what the mapping from a type in ES to a model is.  I think that I'd put this information in your `transform` function, then it's none of Esprit's business what the data actually represents.  So, to anonymise an account:

```python
def anonymise_account(raw):
    account = Account(**raw)
    account.email = anon_emmail(account.email)
    return account.data
```

Reply:
I see what you mean now, the transforms taking charge without esprit passing `type` around works very well. All done now.